staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go

```diff
@@ -85,7 +85,11 @@ func WithLogging(handler http.Handler, pred StacktracePred) http.Handler {
 		rl := newLogged(req, w).StacktraceWhen(pred)
 		req = req.WithContext(context.WithValue(ctx, respLoggerContextKey, rl))
 
-		defer func() { klog.V(3).InfoS("Received HTTP request", rl.LogArgs()...) }()
+		defer func() {
```

Comment:
you could replace defer line with if line

Reply:
or go back to old Log() method with the check and logging inside

Reply:
Do you mean like 
```
if klog.V(3).Enabled() {
			defer func() {klog.InfoS("HTTP", rl.LogArgs()...)}()
		}
```
?

Reply:
yes